src/Symfony/Component/Validator/Constraint.php

```diff
@@ -108,7 +108,7 @@ public function __construct($options = null)
         $defaultOption = $this->getDefaultOption();
         $invalidOptions = [];
         $missingOptions = array_flip((array) $this->getRequiredOptions());
-        $knownOptions = get_object_vars($this);
+        $knownOptions = get_class_vars(static::class);
```

Comment:
Just wondering: do we have tests with static properties, and others with dynamic ones? what behavior do we want for each?

Reply:
There is a new test for static properties which ensures that they cannot be used.

Reply:
Dynamic properties were not supported before and still are not if I don't miss anything.